crates/bevy_reflect/src/reflect.rs

```diff
@@ -93,6 +93,14 @@ pub trait Reflect: Any + Send + Sync {
     /// [`TypeRegistry::get_type_info`]: crate::TypeRegistry::get_type_info
     fn get_represented_type_info(&self) -> Option<&'static TypeInfo>;
 
+    /// Returns the [`TypePath`] implementation for the underlying type.
```

Comment:
Why aren't we just doing `Reflect: DynamicTypePath`? Seems like this just makes the trait more complicated / does more codegen for something that rust can handle for us. Was there rationale here?

Reply:
I think the reason was to match `TypeInfo`. But adding it as a supertrait probably makes things a bit simpler.

Reply:
Another thought is that we may want to have it eventually return the _represented_ type path. In which case, it will need to be changed to return an `Option`.

Reply:
Yeah I think thats the move. I've got a PR prepped.